webapp/channels/src/components/copy_button.tsx

```diff
@@ -41,26 +39,18 @@ const CopyButton: React.FC<Props> = (props: Props) => {
         copyToClipboard(props.content);
     };
 
-    const getId = () => {
-        if (isCopied) {
-            return t('copied.message');
-        }
-        return props.beforeCopyText ? t('copy.text.message') : t('copy.code.message');
-    };
-
-    const getDefaultMessage = () => {
-        if (isCopied) {
-            return props.afterCopyText;
-        }
-        return props.beforeCopyText ?? 'Copy code';
-    };
+    let tooltipMessage;
+    if (isCopied) {
+        tooltipMessage = messages.copied;
+    } else if (props.isForText) {
+        tooltipMessage = messages.copyText;
+    } else {
+        tooltipMessage = messages.copyCode;
+    }
```

Comment:
I see no strong reason to move this into `defineMessages` instead of having one `defineMessage` for each here. The id is not repeated and the object is always expanded so it shouldn't generate re-renders.

0/5 though. Having it outside of the component adds less "unneeded function calls" (not sure if the transpiler gets rid of them) and less objects being created, so less stress for the GC. But I feel having inline the id and default message helps with readability (even though it gets a bit bulkier and uglier XD).

Reply:
Yeah, I am starting to go back in that direction myself, although repeating `defineMessage` 3 times in a row feels like a lot. After all of this is done, I feel like we're going to end up eventually with a custom function called `t` that just takes a proper `MessageDescriptor` so that the FormatJS tooling can read it 😛